src/recipe_dictionary.cpp

```diff
@@ -222,6 +168,37 @@ std::vector<const recipe *> recipe_subset::search( const std::string &txt,
 
     return res;
 }
+static void insert_vector( const recipe_subset &src, recipe_subset &dst, const std::vector<const recipe *> &recipes ) {
```

Comment:
This could be a `recipe_subset` constructor, rather than a static member function (Assuming I'm correct in thinking that you only ever use it to insert into an empty `recipe_subset`).

Reply:
Your assumption is correct, only ever adding to an empty `recipe_subset`. A constructor would take the `const recipe_subset &src` and `const std::vector<const recipe *> &recipes` arguments and transplant the functionality I guess.
`recipe_subset` doesn't currently have a constructor so I would need to define additional constructors (empty/copy at minimum). This is the only situation where we have a possible want for a new constructor, and at the moment would only be useful in the reduce/intersection/difference functions.
Unless there is a broader requirement for new constructors I am reluctant to add one.

Reply:
> Your assumption is correct, only ever adding to an empty `recipe_subset`. A constructor would take the `const recipe_subset &src` and `const std::vector<const recipe *> &recipes` arguments and transplant the functionality I guess.
> `recipe_subset` doesn't currently have a constructor so I would need to define additional constructors (empty/copy at minimum). This is the only situation where we have a possible want for a new constructor, and at the moment would only be useful in the reduce/intersection/difference functions.

You don't need to add a copy constructor; the default continues to exist even if you define others.  And the default constructor is just a one line `recipe_subset() = default;`.

Not vital; I think it would be neater, but that's subjective.

Reply:
ah okay, then my argument against is mostly moot. I'll try out the addition of `recipe_subset() = default;` and `recipe_subset( const recipe_subset &src, const std::vector<const recipe *> );` constructors. Probably is neater, at the very least it should clean up the `reduce`/`intersection`/`difference` functions a little.